sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/throughputControl/ThroughputControlStore.java

```diff
@@ -85,9 +82,12 @@
     private final ConnectionMode connectionMode;
     private final AsyncCache<String, IThroughputContainerController> containerControllerCache;
     private final GlobalEndpointManager globalEndpointManager;
-    private final ConcurrentHashMap<String, Set<ThroughputControlGroup>> groupMapByContainer;
+    private final ConcurrentHashMap<String, Set<ThroughputControlGroupInternal>> groupMapByContainer;
```

Comment:
the other dictionary is thread-safe, shouldn't Set also be ConcurrentHashSet to ensure thread-safety?

Reply:
Make sense.

Previous  I am thinking since currently it is using following pattern for updating the hashes, so it should be fine. 
  this.groupMapByContainer.compute(collectionLink, (key, groupSet) -> {

But I think I see an issue now. Will change in the next iteration.